modules/core/shared/src/main/scala/gem/util/InstantMicros.scala

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2016-2017 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package gem.util
+
+import cats._
+
+import java.time.Instant
+import java.time.temporal.ChronoUnit.MICROS
+
+
+/** InstantMicros wraps a `java.util.Instant` that is truncated to microsecond
+  * resolution.  This allows InstantMicros to roundtrip to/from the database
+  * where timestamps support only microsecond resolution.
+  *
+  * @param toInstant
+  */
+final class InstantMicros private (val toInstant: Instant) extends AnyVal {
+
+  /** Gets the number of seconds from the Java epoch of 1970-01-01T00:00:00Z. */
+  def epochSecond: Long =
+    toInstant.getEpochSecond
+
+  /** Gets the number of microseconds after the start of the second returned
+    * by `epochSecond`.
+    */
+  def µs: Long =
+    toInstant.getNano / 1000l
+
+  /** Converts this instant to the number of milliseconds from the epoch of
+    * 1970-01-01T00:00:00Z.
+    */
+  def toEpochMilli: Long =
+    toInstant.toEpochMilli
+
+  /** Creates an updated instance of InstantMicros by applying the given
+    * function to its wrapped Instant.
+    */
+  def mod(f: Instant => Instant): InstantMicros =
+    InstantMicros.truncate(f(toInstant))
```

Comment:
In scalaz the pointfree version of these functions would be something like `(InstantMicros.truncate _) compose f` which isn't worth the trouble, but do you know if there an nicer syntax for pointfree style in `cats`?